updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -27,6 +27,32 @@ def self.create_from_job_definition(job:, job_definition:)
 
     attr_reader :base_commit_sha, :dependency_files, :dependencies
 
+    # Returns the subset of all project dependencies which are permitted
+    # by the project configuration.
+    def allowed_dependencies
+      @allowed_dependencies ||= dependencies.select { |d| job.allowed_update?(d) }
+    end
+
+    # Returns the subset of all project dependencies which are specifically
+    # requested to be updated by the job definition.
+    def job_dependencies
+      return [] unless job.dependencies&.any?
+      return @job_dependencies if defined? @job_dependencies
```

Comment:
💅 Not significant!

Are there cases where `job.dependencies` would change? It seems 🤏 surprising to have a guard clause that checks state before the `return @ivar if defined? @ivar` clause. I would typically expect that to be a part of the memoization.